This patch fixes a segfault in re_fastputc. This segfault happens in
one-line terminals. It is a backport from upstream netbsd.

See:

	* https://gnats.netbsd.org/cgi-bin/query-pr-single.pl?number=54654
	* https://freshbsd.org/commit/netbsd/src/PxLOa0eNmIl1zBKB

diff -upr libedit-20191025-3.1.orig/src/terminal.c libedit-20191025-3.1/src/terminal.c
--- libedit-20191025-3.1.orig/src/terminal.c	2019-11-14 10:41:42.834954338 +0100
+++ libedit-20191025-3.1/src/terminal.c	2019-11-14 10:43:13.164941706 +0100
@@ -497,7 +497,7 @@ terminal_move_to_line(EditLine *el, int
 	if (where == el->el_cursor.v)
 		return;
 
-	if (where > el->el_terminal.t_size.v) {
+	if (where >= el->el_terminal.t_size.v) {
 #ifdef DEBUG_SCREEN
 		(void) fprintf(el->el_errfile,
 		    "%s: where is ridiculous: %d\r\n", __func__, where);
@@ -647,7 +647,8 @@ terminal_overwrite(EditLine *el, const w
 	if (el->el_cursor.h >= el->el_terminal.t_size.h) {	/* wrap? */
 		if (EL_HAS_AUTO_MARGINS) {	/* yes */
 			el->el_cursor.h = 0;
-			el->el_cursor.v++;
+			if (el->el_cursor.v + 1 < el->el_terminal.t_size.v)
+				el->el_cursor.v++;
 			if (EL_HAS_MAGIC_MARGINS) {
 				/* force the wrap to avoid the "magic"
 				 * situation */
